File: VB/E4882/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Web.Mvc
Imports E4882.Models

Namespace E4882.Controllers
    Public Class HomeController
        Inherits Controller
        Public Function Index() As ActionResult
            ViewData("AutoFilterRowVisible") = True
            Return View(DataRepository.GetData())
        End Function
        Public Function GridViewPartial(ByVal visible As Boolean) As ActionResult
            Return GridViewCorePartial(visible)
        End Function
        Public Function GridViewCustomActionPartial(ByVal visible As Boolean) As ActionResult
            Return GridViewCorePartial(visible)
        End Function
        Private Function GridViewCorePartial(ByVal visible As Boolean) As ActionResult
            ViewData("AutoFilterRowVisible") = visible
            Return PartialView("GridViewPartial", DataRepository.GetData())
        End Function
    End Class
End Namespace

File: CS/E4882/Controllers/HomeController.cs
using System;
using System.Web.Mvc;
using E4882.Models;

namespace E4882.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            ViewData["AutoFilterRowVisible"] = true;
            return View(DataRepository.GetData());
        }
        public ActionResult GridViewPartial(bool visible) {
            return GridViewCorePartial(visible);
        }
        public ActionResult GridViewCustomActionPartial(bool visible) {
            return GridViewCorePartial(visible);
        }
        private ActionResult GridViewCorePartial(bool visible) {
            ViewData["AutoFilterRowVisible"] = visible;
            return PartialView("GridViewPartial", DataRepository.GetData());
        }
    }
}